### PR TITLE
Upgrade `click` dependency

### PR DESCRIPTION
**Pull request by MarcAndre-Rivet**
Updates `click` dependency.

- This allows downstream projects requiring `celery>=5.3` to continue using this project, as Celery>=5.3 requires `click>=8.1.2` / causing an incompatible version matchup in Poetry
- Dropping py36 should not be of consequence as it's been past EOL for over a year now
- Testing against more recent versions of Python (3.10+) required an update to `pytest` as well

### CODE REVIEW

**Comment by MarcAndre-Rivet** [comment]
This gh rule will need updating as this will never be executed
image

**Review by tacrocha — DISMISSED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -12,13 +12,13 @@ authors = [
 license = "MIT"
 
 [tool.poetry.dependencies]
-python = "^3.6"
+python = ">=3.7,<3.12"
```
Comment (MarcAndre-Rivet) [inline pyproject.toml:15]: Updating click drops py3.6 support. The project also doesn't work with py3.12 as per test run as-is, hence the updated version range. We can always revisit as necessary.

File: .github/workflows/pythonpackage.yml [hunk 2]
```diff
@@ -10,8 +10,11 @@ jobs:
       matrix:
         os-python-version:
           [
-            { "os": "ubuntu-18.04", "python": 3.6 },
-            { "os": "ubuntu-latest", "python": 3.7 },
+            { "os": "ubuntu-latest", "python": "3.7" },
+            { "os": "ubuntu-latest", "python": "3.8" },
+            { "os": "ubuntu-latest", "python": "3.9" },
+            { "os": "ubuntu-latest", "python": "3.10" },
+            { "os": "ubuntu-latest", "python": "3.11" },
```
Comment (MarcAndre-Rivet) [inline .github/workflows/pythonpackage.yml:17]: testing the full range of versions supported

**Review by codingtony-candid — APPROVED** [review]
lgtm

**Pull request merged**